core/arch/arm/mm/tee_pager.c

```diff
@@ -532,9 +532,14 @@ void tee_pager_add_core_area(vaddr_t base, enum tee_pager_area_type type,
 	uint32_t flags = 0;
 	size_t fobj_pgoffs = 0;
 	vaddr_t b = base;
-	size_t s = fobj->num_pages * SMALL_PAGE_SIZE;
+	size_t s;
 	size_t s2 = 0;
 
+	if (!fobj)
+		panic();
+
+	s = fobj->num_pages * SMALL_PAGE_SIZE;
```

Comment:
This change is not really needed, if `fobj` is NULL, Core would already panic on a data abort (seg fault).
Moreover, we don't expected a NULL `fobj` at entry.
Maybe we can use `assert(fobj);`, not `if (fobj) panic();` here, not nicely catch the issue in debug mode but I don't think it would add much.
I suggest to discard this change.

Reply:
the if !fobj check existed since before - i just moved it up so that it actually can have an effect...

Reply:
you're right, sorry. I missed the removed lines below.
Actually, should we panic() here, assert fobj is not NULL or should we simply remove the `if (!fobj) panic();` and let `size_t s = fobj->num_pages * SMALL_PAGE_SIZE;` trig a segfault caught by Core abort handler?

I would be in favor of the later and only remove lines 557..559.
@jenswi-linaro? @jforissier? and all?

Do we really want to assert all arguments in all internal or exported function?
There is no unpredictable behavior here on fault NULL reference: Core cleanly treats segfaults, possibly panicking when needed.

Reply:
static code analysis will highlight the risk or nullptr deref unless there is an assert that shows we are ok with not expecting null arg...

Reply:
> Do we really want to assert all arguments in all internal or exported functions?

No, we don't want to add mindless asserts everywhere.

> static code analysis will highlight the risk or nullptr deref unless there is an assert that shows we are ok with not expecting null arg...

￼That is very unspecific. Which tool are we talking about? To me this sounds like asserts are used to silence warnings in that particular tool. Does this tool take only possible code paths into account like Coverity if configured correctly?

Reply:
The tool is Cppcheck 1.89 and it does not check for currently possible code paths, its focus is to make exported functions not crash.

Since the NULL check existed before my commit (i just moved it) i think it is better this way, but im fine with removing the null check and getting the panic as a side-effect if that is how other null ptr deref is managed.